monkey/agent_plugins/credentials_collectors/chrome/src/windows_credentials_database_selector.py

```diff
@@ -1,10 +1,155 @@
-from pathlib import PurePath
-from typing import Sequence
+# NOTE: This only tries stealing credentials from the current user,
+#       so no DPAPI logic is implemented yet. That requires more research.
+#       We can decide to do this now or in a new version of the plugin.
+#       (WinAPI is used for current user, DPAPI is used otherwise.)
+
+
+import base64
+import getpass
+import json
+import logging
+from pathlib import Path, PureWindowsPath
+from typing import Dict, Optional, Set, Tuple
+
+from .windows_utils import Win32CryptUnprotectData
+
+logger = logging.getLogger(__name__)
+
+DRIVE = "C"
+LOCAL_APPDATA = "{drive}:\\Users\\{user}\\AppData\\Local"
+
+WINDOWS_BROWSERS = {
+    "Chromium Edge": "{local_appdata}\\Microsoft\\Edge\\User Data",
+    "Google Chrome": "{local_appdata}\\Google\\Chrome\\User Data",
+}
 
 
 class WindowsCredentialsDatabaseSelector:
     def __init__(self):
-        pass
+        user = getpass.getuser()
+        local_appdata = LOCAL_APPDATA.format(drive=DRIVE, user=user)
+
+        self._browsers: Dict[str, PureWindowsPath] = {}
+        for browser_name, browser_directory in WINDOWS_BROWSERS.items():
+            self._browsers[browser_name] = PureWindowsPath(
+                browser_directory.format(local_appdata=local_appdata)
+            )
+
+    def __call__(self) -> Set[Tuple[PureWindowsPath, Optional[bytes]]]:
```

Comment:
Consider adding a dataclass or NamedTuple type that contains dbpath and decryption key and returning that instead of a tuple.